osbs/utils.py

```diff
@@ -289,6 +289,10 @@ def get_time_from_rfc3339(rfc3339):
         return timegm(time_tuple)
 
 
+
+VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[a-z0-9]')
```

Comment:
BTW `strip` and `rstrip` is not sufficient. Need something like `re.sub` to ensure proper replacement when the repo or branch name ends in both `'-'` and `'.'`:

```python
>>> repo = '---...--..blah..---..--'
>>> re.sub(r'^(\.|\-)*', '', repo)
'blah..---..--'
>>> re.sub(r'(\.|\-)*$', '', repo)
'---...--..blah'
```

Definitely a corner case, but I'd rather have this fixed in a way that won't bite us again.